addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/handler/AirQualityHandler.java

```diff
@@ -0,0 +1,303 @@
+     * Get new data from aqicn.org service
+     *
+     * @return {boolean}
+     */
+    private synchronized boolean updateAirQualityData() {
+        AirQualityConfiguration config = getConfigAs(AirQualityConfiguration.class);
+        aqiData = getAirQualityData(StringUtils.trimToEmpty(config.location));
+        return aqiData != null;
+    }
+
+    /**
+     * Request new air quality data to the aqicn.org service
+     *
+     * @param location geo-coordinates from config
+     *
+     * @return the air quality data object mapping the JSON response or null in case of error
+     */
+    private AirQualityJsonResponse getAirQualityData(String location) {
+        AirQualityJsonResponse result = null;
+        boolean resultOk = false;
+        String errorMsg = null;
+
+        try {
+
+            // Build a valid URL for the aqicn.org service
+            AirQualityConfiguration config = getConfigAs(AirQualityConfiguration.class);
+
+            String geoStr = location.replace(" ", "").replace(",", ";").trim();
+
+            String urlStr = URL.replace("%APIKEY%", StringUtils.trimToEmpty(config.apikey));
+            urlStr = urlStr.replace("%LOCATION%", geoStr);
+            logger.debug("URL = {}", urlStr);
+
+            // Run the HTTP request and get the JSON response from aqicn.org
+            URL url = new URL(urlStr);
+            URLConnection connection = url.openConnection();
+            String repsonse = IOUtils.toString(connection.getInputStream());
+            logger.debug("aqiData = {}", repsonse);
+
+            // Map the JSON response to an object
+            result = gson.fromJson(repsonse, AirQualityJsonResponse.class);
+            if (result == null) {
+                errorMsg = "no data returned";
+            } else if (result.getStatus() == "error") {
+                errorMsg = result.getData().toString();
+            } else {
+                resultOk = true;
+
+                if (result.getData() == null) {
+                    resultOk = false;
+                    errorMsg = "missing data sub-object";
+                }
+            }
+
+            if (!resultOk) {
+                logger.warn("Error in aqicn.org (Air Quality) response: {}", errorMsg);
+            }
+
+        } catch (MalformedURLException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Constructed url is not valid: {}", errorMsg);
+        } catch (IOException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Error running aqicn.org (Air Quality) request: {}", errorMsg);
+        } catch (JsonSyntaxException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Error parsing aqicn.org (Air Quality) response: {}", errorMsg);
+        }
+
+        // Update the thing status
+        if (resultOk && result != null) {
+            String attributions = result.getData().getAttributions();
+            logger.info("Received AirQuality data from aqicn.org. {}", attributions);
```

Comment:
Do we need this result attribution (at info level) at each and every data query? Maybe it is good enough to show it only on Thing initialization?